### ISSUE TITLE
Issue with V2.0 CAD

### ISSUE THREAD

**Issue by maxsu** [Nate711/StanfordDoggoProject#34]
Hi all,

It looks like Doggo's v2 CAD model is experiencing some issues - From the web viewer:

![image](https://user-images.githubusercontent.com/25123/63017860-3bb4b000-be4c-11e9-8651-0253a448218f.png)

![image](https://user-images.githubusercontent.com/25123/63017923-630b7d00-be4c-11e9-902b-3e8efc0cb620.png)

Description:
- There are left-translated copies of the right motor & shaft assemblies and the forward sheet metal component
- Affected items are translated forward relative to the side plates and legs
- Affected items are canted WRT the right side plate, so that their right mating faces are vertical
- All other parts and sub-assemblies appear properly positioned 

![image](https://user-images.githubusercontent.com/25123/63018787-8a634980-be4e-11e9-9968-8f73403199ec.png)
![image](https://user-images.githubusercontent.com/25123/63018793-8c2d0d00-be4e-11e9-9458-b607bfd51053.png)

The DXF export of the v2 cad shows the same issues. I could not check the file in A360 due to the "Access Denied" issue.

I could eventually rebuild the model's assembly tree if necessary. Before I do that:
- Can anyone else confirm the issue?
- If the issue is more than an A360 cache glitch, is a repaired model available?

**Comment by mseedman** [comment]
I can confirm the issue.  Michael

**Comment by Nate711** [comment]
Can you try checking again? I re-computed the model. I really have no idea why Fusion does this....

**Comment by maxsu** [comment]
Thanks for the update - the assembly looks good now.

The "Access Denied" issue persists.  It was first mentioned in #31, where @mseedman had trouble with the Leg Linkage file. Currently, the Single Plate Design file blocks the import process:

![image](https://user-images.githubusercontent.com/25123/63818230-f6d15480-c8f4-11e9-99e5-efe00d0293ed.png)

**TLDR;;** I'd like to do a test: Could you create and post a brand new public link here?

### Rationale

This [forum post](https://forums.autodesk.com/t5/fusion-360-support/file-could-not-be-uploaded-due-to-access-denied/td-p/6683883) lists possible `access denied` causes:

1. The file could be archived
2. The project may be inactive
3. There may be a A360 [team hub](https://myhub.autodesk360.com) issue
4. The current A360 build could be bad
5. General access rights might not have propagated to a new user

I can rule out the last issue - I have no problem accessing my own cloud files, and I can import most doggo files. To partially rule out issue 4, I've done a fresh install with the [newest fusion build]( https://manage.autodesk.com/cep/#products-services/), making sure to use the [f360 cleaner](https://knowledge.autodesk.com/support/fusion-360/troubleshooting/caas/sfdcarticles/sfdcarticles/How-to-do-an-automatic-Clean-Uninstall-for-Autodesk-Fusion-360.html), and  run the installer in administrator mode.

A360 hub doesn't have public teams/projects, or any explicit per-file access controls. Further, given that none of the components have been archived, this is not quite a team issue, but a public link issue. Therefore, I tentatively advance the theory that the link itself is corrupt.

Cy has written a quick [guide ](http://tech.cyborg5.com/2018/07/28/workaround-for-saving-fusion-360-projects-with-linked-components/)on public links for open source sharing - I'll assume the design admin obtains the public link directly from the top level assembly.

Lets suppose that link references an independently created object that **includes references to all descendant parts** - this is essentially the cloud counterpart of `Full Doggo Assembly.f3z/DesignDescription.json`. If these references go stale - for whatever reason - then the access denied issue could happen. Likewise, it's possible that  regenerating the public link would fix this condition.

**Conclusion/Next Actions**: It would be easy to generate and test a new public link; it's feasible that a refreshed link will fix the reference issue.

- If it fixes the issue -> consult autodesk support to prevent future regressions
- If issue persists -> consult autodesk support for a fix

**Comment by maxsu** [comment]
I've made a [post](https://forums.autodesk.com/t5/fusion-360-support/stanford-doggo-public-link-upload-issue/td-p/9000575) on the fusion 360 forums - the community may shed some light on the upload issue.

I've also found a temporary workaround. cc: @mseedman - you may find this useful.

### Workaround: Import an Inventor 2018 Model

1. Open doggo v2's [public link](https://a360.co/2CPZoto)
2. Request, Download and Unzip the Inventor 2018 model
3. Open `Full Doggo Assembly.iam` in Fusion 360
4. Fusion will prompt you to bulk-select the rest of Doggo's files

This is useful for locally inspecting the model. One issue: this doesn't add individual parts or assemblies to your project - it bundles all of doggo into a single top level assembly.

**Comment by maxsu** [comment]
Good news! Autodesk support team has provided a provisional fix for the "access denied issue" (see [this thread](https://forums.autodesk.com/t5/fusion-360-support/stanford-doggo-public-link-upload-issue/m-p/9006809/highlight/true#M65178)).  The `f3z` file contained in Jeff.Strather's `Full Doggo Assembly_Repaired.zip` loads without causing the "access denied" error.

Upon inspection, it looks like they removed the two simstudies from the project's `.f3z` file. Jeff has requested a member of the Doggo team to discuss steps to fix the original model (see the thread).


(Security note: Jeff appears to be an official member of the Autodesk team, and the file [checks out](https://www.virustotal.com/gui/file/7b64bfddaeec573c237a05f7cec86e4d40eb3dcb6a20d5a1e902fcf30e65f8c2/detection) on VirusTotal. Use your discretion here, and wait for the definitive fix if there are any concerns.)

CC: @CrazyAZ ,  @mseedman
